coreos-meta-translator/trans.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+from argparse import ArgumentParser, FileType
+from datetime import datetime
+import json
+import os
+
+
+def ensureDup(inp, out, inp_key, out_key):
+    if out.get(out_key, None) == None:
+        out[out_key] = inp.get(inp_key)
+        return out
+    if out.get(out_key) != inp.get(inp_key):
+        print("Input Files do not appear to be for the same release")
+        raise SystemExit(1)
+    return out
+
+def url_builder(stream, version, arch, path):
+    return f"https://fcos-builds.s3.amazonaws.com/prod/streams/{stream}/builds/{version}/{arch}/{path}"
```

Comment:
In general, I don't think we should use the `amazonaws.com` at all (and instead `https://builds.coreos.fedoraproject.org/`). Both for encapsulation, but also because `builds.c.fp.o` goes through CloudFront.